prepare access list for eth_call

yihuang opened the pull request:
Closes: #335

<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->


## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

FrancoCRO commented:
Checked that this PR resolves the problem

fedekunze commented on app/ante/ante.go:
```diff
@@ -70,7 +70,6 @@ func NewAnteHandler(
 						NewEthNonceVerificationDecorator(ak),
 						NewEthGasConsumeDecorator(ak, bankKeeper, evmKeeper),
 						NewCanTransferDecorator(evmKeeper),
-						NewAccessListDecorator(evmKeeper),
```
what is the issue with this decorator? the logic is pretty much the same one

yihuang replied:
The issue is ante handler is not executed for `eth_call`.

fedekunze replied:
ok that makes sense. Why aren't we using the `Simulate` gRPC method again? then we could easily solve this issue from arising in the future if we add more logic to the ante handler

yihuang replied:
`Simulate` don't support execute on historical block height, and cosmos-sdk team seems reluctant to support that https://github.com/cosmos/cosmos-sdk/issues/9636#issuecomment-874687473.

yihuang replied:
Also to support `eth_call` in ante, we need to further adapt the ante handler logic, since in `eth_call` the `MsgEthereumTx` is not signed, and `from` could be empty.

fedekunze commented:
we should add additional notes on the code about this logic but will merge this to speed up things. Please add another PR describing the rationale

yihuang commented:
> we should add additional notes on the code about this logic but will merge this to speed up things. Please add another PR describing the rationale

Sure, I just added a comment, also rebased to recent main branch.
